src/dependency-builder/build-tree.js

```diff
@@ -387,6 +388,7 @@ export default async function getDependencyTree(baseDir: string, consumerPath: s
       }
     });
   }
+  groups.packages = packageJson ? groups.packages.filter(packageName => !(packageName in packageJson.dependencies)) : groups.packages;
```

Comment:
this search should be recursive starting from the component root dir and bubble up to the consumer. 
like the findPath function above.
let's talk about it